sormas-ui/src/main/java/de/symeda/sormas/ui/utils/ExternalJournalUtil.java

```diff
@@ -29,19 +32,93 @@
 import java.nio.charset.StandardCharsets;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.Optional;
 
 public class ExternalJournalUtil {
 
 	private final static ExternalJournalFacade externalJournalFacade = FacadeProvider.getExternalJournalFacade();
 
+	/**
+	 * Creates a Button to be added wherever a patient needs to be managed by an external journal.
+	 * If no external journal is enabled with the current settings, an empty optional is returned.
+	 * If the person is not registered in the external journal, a create account/register button is returned
+	 * If the person is registered, a button is returned which opens a popup with further options.
+	 * @param person person to be managed by the external journal
+	 * @return Optional containing appropriate Button
+	 */
+	public static Optional<Button> getExternalJournalUiButton(PersonDto person) {
+		if (FacadeProvider.getConfigFacade().getSymptomJournalConfig().isActive()) {
+			if (person.isEnrolledInExternalJournal()) {
+				return Optional.of(createPiaOptionsButton(person));
+			} else {
+				return Optional.of(createPiaRegisterButton(person));
+			}
+		}
+		else if (FacadeProvider.getConfigFacade().getPatientDiaryConfig().isActive()) {
+			if (person.isEnrolledInExternalJournal()) {
+				return Optional.of(createClimedoOptionsButton(person));
+			} else {
+				return Optional.of(createClimedoRegisterButton(person));
+			}
+
+		}
+		return Optional.empty();
+	}
+
+	private static Button createPiaOptionsButton(PersonDto person) {
+		VerticalLayout popupLayout = new VerticalLayout();
+		popupLayout.setSpacing(true);
+		popupLayout.setMargin(true);
+		popupLayout.addStyleName(CssStyles.LAYOUT_MINIMAL);
+		// TODO: implement cancel for PIA
+		Button.ClickListener cancelListener = clickEvent -> {};
+		Button.ClickListener openListener = clickEvent -> openSymptomJournalWindow(person);
+		PopupButton ediaryButton = ButtonHelper.createPopupButton(I18nProperties.getCaption(Captions.piaOptionsButton), popupLayout, ValoTheme.BUTTON_PRIMARY);
+		Button cancelButton = ButtonHelper.createButton(I18nProperties.getCaption(Captions.cancelExternalFollowUpButton), cancelListener, ValoTheme.BUTTON_PRIMARY);
+		Button openButton = ButtonHelper.createButton(I18nProperties.getCaption(Captions.openInPiaButton), openListener, ValoTheme.BUTTON_PRIMARY);
+		popupLayout.addComponent(cancelButton);
+		popupLayout.addComponent(openButton);
+		return ediaryButton;
+	}
+
+
+	private static Button createPiaRegisterButton(PersonDto person) {
```

Comment:
Also methods could be named more generically and optionally specify in the JavaDoc that it's used by PIA, but I would do it only for public methods